### PR TITLE
1719690: Update to message formatting

### PR DESCRIPTION
**Pull request by wottop**

### CODE REVIEW

**Review by ptoscano — CHANGES_REQUESTED** [review]
Ideally, we'd fix things in `main` first, to avoid forgetting to forwardport them.

File: src/subscription_manager/certlib.py [hunk 1]
```diff
@@ -55,7 +55,10 @@ def log_entry(self):
     def format_exceptions(self):
         buf = ''
         for e in self._exceptions:
-            buf += ' '.join(str(e).split('-')[1:]).strip()
+            if len(str(e).split('-')) == 1:
+                buf += str(e).split('-')[0].strip()
+            else:
+                buf += '-'.join(str(e).split('-')[1:]).strip()
```
[the remark this answers is not in the document]
Comment (wottop) [reply src/subscription_manager/certlib.py]: Yes. Note that I also had to fix the case where there is no '-' in the error message.

File: src/subscription_manager/certlib.py [hunk 2]
```diff
@@ -55,7 +55,10 @@ def log_entry(self):
     def format_exceptions(self):
         buf = ''
         for e in self._exceptions:
-            buf += ' '.join(str(e).split('-')[1:]).strip()
+            if '-' not in str(e):
+                buf += str(e)
+            else:
+                buf += ''.join(str(e).split('-', maxsplit=1)[1:]).strip()
```
Comment (ptoscano) [inline src/subscription_manager/certlib.py:61]: considering that `split()` always returns a list, which in this case has either 1 (no '-') or 2 elements (`maxsplit=1`), then we can simplify this like this:
```python
    buf += str(e).split('-', maxsplit=1)[-1].strip()
```
no need for checking the character presence, and no need for `join()`

**Review by ptoscano — CHANGES_REQUESTED** [review]
One more fixup, and it should be ok hopefully :)

I traced this back to when it was introduced, to know why it actually was done: I found commit 6c3a03f54c2e650fa7fd7f18d84f71aabdd75d89 for this, which unfortunately does not explain why the message had to be mungled that way...

**Review by ptoscano — APPROVED** [review]
LGTM, thanks for the updates!

Please remember to forwardport it to `main`, to avoid losing it.

**Pull request merged**